R/utils-read.R

```diff
@@ -635,12 +635,14 @@ readMeanGageHeights<- function(reportObject){
 #' @description get the list of readings attached to a report. Will include a year+month field as a month identifier for each record.
 #' @param reportObject the full JSON report object
 #' @param filter optional filter to restrict to reading types (reference, crestStage, or waterMark)
+#' @param derivation [DEFAULT: "primary"] The derivation chain position to get the readings from 
 #' @return data frame of reading records
-readReadings <- function(reportObject, filter="") {
-  time <- as.POSIXct(strptime(reportObject[['readings']][['time']], "%FT%T"))
-  value <- as.numeric(reportObject[['readings']][['value']])
-  type <- reportObject[['readings']][['type']]
-  uncertainty <- as.numeric(reportObject[['readings']][['uncertainty']])
+readReadings <- function(reportObject, filter="", derivation="primary") {
+  jsonField <- paste0(derivation, "Readings")
```

Comment:
I don't think we should do this hidden convention and instead force all callers to pass in the full field name. I'm not sure "derivation" is the correct term. I'd go with "readingsFieldName" to be explicit.